tapa/common/graph.py

```diff
@@ -86,3 +93,101 @@ def get_flatten_graph(self) -> "Graph":
         }
 
         return Graph(self.name, new_obj)
+
+    def get_floorplan_slot(
+        self, slot_name: str, tasks_in_slot: list[str], top: Base
+    ) -> TaskDefinition:
+        """Return the task grouping the tasks floorplanned in the given slot."""
+        # Construct obj of the slot by modifying the top task
+        new_obj = self.get_top_task_def().to_dict()
+        new_obj["level"] = "upper"
+
+        # Find all task instances
+        insts = self.get_top_task_inst().get_subtasks_insts()
+        for inst in insts:
+            # make sure top has been flattened
+            assert isinstance(inst.definition, TaskDefinition)
+            assert inst.definition.get_level() == TaskDefinition.Level.LEAF, (
+                "Top task must be flattened for floorplanning"
+            )
+
+        # filter out tasks that are not in the slot
+        new_insts = [inst for inst in insts if inst.name in tasks_in_slot]
+
+        assert new_insts, [inst.name for inst in insts]
+
+        # obj['tasks']:
+        # construct the task insts of the slot
+        new_obj["tasks"] = defaultdict(list)
+        for inst in new_insts:
+            new_obj["tasks"][inst.definition.name].append(inst.to_dict())
+
+        # obj['fifos']:
+        # construct the fifos of the slot
+        fifos = self.get_top_task_inst().get_interconnect_insts()
+        new_fifos: list[InterconnectInstance] = []
+        fifo_ports: list[str] = []
+        for fifo in fifos:
+            fifo_obj = fifo.to_dict()
+            src = fifo_obj["consumed_by"]
+            dst = fifo_obj["produced_by"]
+            # For fifo connecting task insts inside the slot, keep it
+            if (
+                INST_NAME.format(src[0], src[1]) in tasks_in_slot
```

Comment:
Can you use `tapa.util.get_instance_name(src)` instead of `INST_NAME.format`? It's important that we maintain a single source of truth (per language, at least) for the naming conventions.